### PR TITLE
Light Sorc improvement, Kill process on stop and LK run improvements

### PR DESCRIPTION
**Pull request by nullptr-ref**
- Change Lightening sorc max range (to 10 from 15) - This improves the overall flow of the lightening sorc as it sometimes tries to attack mobs outside of the nova range.
- Add process kill on supervisor stop.
- Added a check for keys in inventory before trying to open locked chests in LK runs.  mentioned in #145 . We probably need to add some kind of max tries in the InteractObject logic to fix this globally.

### CODE REVIEW

File: internal/run/lower_kurast_chests.go [hunk 1]
```diff
@@ -55,24 +57,32 @@ func (a LowerKurastChest) BuildActions() []action.Action {
...
-									for _, obj := range d.Objects {
-										isSameObj := obj.Name == chest.Name && obj.Position.X == chest.Position.X && obj.Position.Y == chest.Position.Y
-
-										if isSameObj && !obj.Selectable {
-											return true
+
+							_, keyFound := d.Items.Find("Key", item.LocationInventory)
+
+							if chest.InteractType == object.InteractTypeLocked && !keyFound {
+								//Skip the chest if its locked and we don't have keys in our inventory
```
Comment (theksk23) [inline internal/run/lower_kurast_chests.go:64]: kind of the same question I had for https://github.com/hectorgimenez/koolo/pull/148
You implement this here in a specific run directly: Any Idea about how we can get a global setting like maybe `Character.OpenChests`, and prevent putting this logic into every run we implement?

For example Tomb runs do want to open chests, and so run into this problem, too.
/whishlist feel free to close this comment without resolving, thanks!

Comment (nullptr-ref) [reply internal/run/lower_kurast_chests.go:64]: The way I see it, due to the way its written, we will probably need a new method to replace InteractObject. If we create a method like InteractChest that has this check inside and it gets used instead of interactobject in all scripts then maybe, yea. But for now, this check is the only option we have. 

Another thought is to implement the openchest logic inside the clear room function with the setting you mentioned that will solve us having to manually implement openchests in every single script except the ones that only need to open chests where manual interactions will be required.

@hectorgimenez your thoughts on this?

Comment (hectorgimenez) [reply internal/run/lower_kurast_chests.go:64]: I like the OpenChest action idea, we can embed this logic inside and just call InteractObject step if required or return nil otherwise. I don't know if it will be 1/1 replacement for all the usecases.

File: internal/supervisor.go [hunk 2]
```diff
@@ -65,7 +67,16 @@ func (s *baseSupervisor) Stop() {
 	}
 
 	s.c.Injector.Unload()
+	process, err := os.FindProcess(int(s.c.Reader.Process.GetPID()))
```
Comment (hectorgimenez) [inline internal/supervisor.go:70]: I don't know if this is something we want to do? For example when I stop it, I like to go to the stash and see what I have there. But regardless if we want to start this discussion or not, I think we should do it in a different PR to not mix stuff

File: internal/run/lower_kurast_chests.go (hunk 1 above)
Comment (nullptr-ref) [reply internal/run/lower_kurast_chests.go:64]: Just to be on the same page, you're talking about implementing the openchests inside of the clear_area function, correct? 

We can also implement another function to open chests (like interactobject) to have the same check inside that can be used for scripts like lower kurast chest farm. Not sure if there are any other use-cases but might be nice to have, I suppose.

Comment (hectorgimenez) [reply internal/run/lower_kurast_chests.go:64]: oh nono, I meant the second option, implement a new action like interactobject but doing the check inside
